src/WCSimPrimaryGeneratorAction.cc

```diff
@@ -749,6 +756,60 @@ void WCSimPrimaryGeneratorAction::GeneratePrimaries(G4Event* anEvent)
       SetBeamDir(dir);
       SetBeamPDG(pdg);
     }
+
+  else if (useLightInjectorEvt)
+    {
+
+        // Initialise the light injector once per sim. 
+        // This will get LI settings (position, direction, etc) from the db
+        // and produce a 3D histogram of the LI profile.
+        if ( !LIGen ) {
+            LIGen = new WCSimLIGen();
+            LIGen->SetPhotonMode(photonMode);
+            LIGen->ReadFromDatabase(injectorType,injectorIdx,injectorFilename);
+        }
+
+        // Generate the required number of photons with
+        // directions distributed as per the LI profile        
+        // and populate the G4Event
+
+        G4ThreeVector vtx = LIGen->GetInjectorPosition();
+        G4ThreeVector dir = LIGen->GetInjectorDirection();
+        G4int pdg = 0;
+        G4double E = LIGen->GetPhotonEnergy();
+        LIGen->GeneratePhotons(anEvent,nphotons);
+
+        // Save the mc properties for validation
+        // Limit of <900 photons due to size of array in jhfntuple
+        /*
```

Comment:
Can this block be deleted?

I would expect all LI photons you generate to be in the output file, in the track store. Is that not the case? Is there a benefit of saving this information too? (Naively I'd presume all photons have the same start point, energy, PDG... it's just the direction that's different photon by photon - and that info [is in WCSimRootTrack](https://github.com/WCSim/WCSim/blob/develop/include/WCSimRootEvent.hh#L38))